packit_service/worker/helpers/open_scan_hub.py

```diff
@@ -278,3 +280,218 @@ def get_srpm_build(commit_sha):
+
+                    if not output:
+                        raise OSHNoFeedback("Something went wrong, skipping the reporting.")
+
+                    logger.info("Scan submitted successfully.")
+
+                    response_dict = self.parse_dict_from_output(output)
+
+                    logger.debug(f"Parsed dict from output: {response_dict} ")
+
+                    if id := response_dict.get("id"):
+                        scan.task_id = id
+                        scan.status = OSHScanStatus.pending
+                    else:
+                        raise OSHNoFeedback(
+                            "It was not possible to get the Open Scan Hub task_id "
+                            "from the response.",
+                        )
+
+                    if not (url := response_dict.get("url")):
+                        err_msg = "It was not possible to get the task URL from the OSH response."
+                        raise OSHNoFeedback(err_msg)
+                    scan.url = url
+
+                    self.report(
+                        state=BaseCommitStatus.running,
+                        description=(
+                            "Scan in OpenScanHub submitted successfully. "
+                            "Check the URL for more details."
+                        ),
+                        url=get_openscanhub_info_url(scan.id),
+                        links_to_external_services={"OpenScanHub task": url},
+                    )
+            except IntegrityError as ex:
+                logger.info(f"OpenScanHub already submitted: {ex}")
+            except OSHNoFeedback as ex:
+                logger.info(f"OpenScanHub feedback missing: {ex}")
+                self.report(
+                    state=BaseCommitStatus.neutral,
+                    description=err_msg,
+                    url=build_dashboard_url,
+                )
+
+    def report(
+        self,
+        state: BaseCommitStatus,
+        description: str,
+        url: str,
+        links_to_external_services: Optional[dict[str, str]] = None,
+    ):
+        check_name = "osh-diff-scan:fedora-rawhide-x86_64"
+        if identifier := self.koji_build_helper.job_config.identifier:
+            check_name += f":{identifier}"
+        self.koji_build_helper._report(
+            state=state,
+            description=description,
+            url=url,
+            check_names=[check_name],
+            markdown_content=OPEN_SCAN_HUB_FEATURE_DESCRIPTION,
+            links_to_external_services=links_to_external_services,
+        )
+
+    def find_base_build_job_id(self) -> Optional[int]:
+        """
+        Find the job in the config that can provide the base build for the scan
+        (with `commit` trigger and same branch configured as the target PR branch).
+        """
+        # TODO: Alternative we can specify the id of the koji build as base
+        # We can take this approach if Packit does not have a base build in its database.
```

Comment:
Fedora CI is supposed to work for all Fedora packages, while Packit downstream automation will probably never work for all (or even the majority) of them, so I don't think we can realistically expect to have base builds in the DB.